src/app/forms/components/required-standard-select/required-standard-select.component.ts

```diff
@@ -1,9 +1,98 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DefectGETIVA, RequiredStandard, SectionIVA } from '@dvsa/cvs-type-definitions/types/iva/defects/get';
+import { INSPECTION_TYPE } from '@models/test-results/test-result-required-standard.model';
+import { Store } from '@ngrx/store';
+import { RequiredStandardState } from '@store/required-standards/reducers/required-standards.reducer';
+import { getRequiredStandardsState } from '@store/required-standards/selectors/required-standards.selector';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'app-required-standard-select',
   templateUrl: './required-standard-select.component.html',
+  styleUrls: ['./required-standard-select.component.scss'],
 })
-export class RequiredStandardSelectComponent {
+export class RequiredStandardSelectComponent implements OnInit, OnDestroy {
 
+  requiredStandards?: SectionIVA[];
+  normalAndBasic?: boolean;
+  isEditing = false;
+  selectedInspectionType?: INSPECTION_TYPE;
+  selectedSection?: SectionIVA;
+  selectedRequiredStandard?: RequiredStandard;
+  basicAndNormalRequiredStandards?: DefectGETIVA;
+
+  onDestroy$ = new Subject();
+
+  constructor(
+    private requiredStandardsStore: Store<RequiredStandardState>,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) {}
+
+  ngOnInit(): void {
+    this.requiredStandardsStore.select(getRequiredStandardsState).subscribe((requiredStandards) => {
```

Comment:
```suggestion
    this.requiredStandardsStore.select(getRequiredStandardsState).pipe(takeUntil(this.onDestroy$)).subscribe((requiredStandards) => {
```

Want to close this subscription when component destroys to avoid memory leak.

Reply:
thanks!